.github/workflows/build.yml

```diff
@@ -135,26 +135,41 @@ jobs:
           [[ -n $runner ]] && printf 'CROSS_RUNNER=%s\n' "$runner" >> $GITHUB_ENV
         fi
 
+    - name: Build wasmtime library
+      if: ${{ !matrix.use-cross && contains(matrix.features, 'wasm') }}
+      run: |
+        git clone --depth=1 --filter=blob:none https://github.com/bytecodealliance/wasmtime
+        cd wasmtime
+        cmake -S crates/c-api -B target/c-api --install-prefix "$PWD/artifacts"
+        cmake --build target/c-api && cmake --install target/c-api
+        printf 'CMAKE_PREFIX_PATH=%s\n' "$PWD/artifacts" >> $GITHUB_ENV
+
     - name: Build C library (make)
       if: ${{ runner.os != 'Windows' }}
       run: make.sh -j CFLAGS="$CFLAGS"
       env:
-        CFLAGS: -g -Werror -Wall -Wextra -Wshadow -Wno-unused-parameter -pedantic
+        CFLAGS: -g -Werror -Wall -Wextra -Wshadow -Wpedantic -Werror=incompatible-pointer-types
 
     - name: Build C library (CMake)
       if: ${{ !matrix.use-cross }}
-      run: |-
+      run: |
         cmake -S lib -B build \
           -DBUILD_SHARED_LIBS=OFF \
           -DCMAKE_BUILD_TYPE=Debug \
-          -DCMAKE_COMPILE_WARNING_AS_ERROR=ON
+          -DCMAKE_C_COMPILER=clang \
```

Comment:
Using clang to avoid an [incorrect GCC diagnostic](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=83584).